lib/utils/compose_ts.ts

```diff
@@ -14,28 +14,98 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import * as _ from 'lodash';
+import * as Bluebird from 'bluebird';
+import * as tar from 'tar-stream';
+import { Readable } from 'stream';
 
-import { RegistrySecrets } from 'resin-multibuild';
-import { Pack } from 'tar-stream';
+import * as MultiBuild from 'resin-multibuild';
+import { Composition } from 'resin-compose-parse';
+
+import { DeviceInfo } from './device/api';
+import Logger = require('./logger');
 
 /**
- * Return a callback function that takes a tar-stream Pack object as argument
- * and uses it to add the '.balena/registry-secrets.json' metadata file that
- * contains usernames and passwords for private docker registries. The builder
- * will remove the file from the tar stream and use the secrets to pull base
- * images from users' private registries.
- * @param registrySecrets JS object containing registry usernames and passwords
- * @returns A callback function, or undefined if registrySecrets is empty
+ * Create a BuildTask array of "resolved build tasks" by calling multibuild
+ * .splitBuildStream() and performResolution(), and add build stream error
+ * handlers and debug logging.
+ * Both `balena build` and `balena deploy` call this function.
  */
-export function getTarStreamCallbackForRegistrySecrets(
```

Comment:
The `getTarStreamCallbackForRegistrySecrets` function was deleted because it was an unused duplicate of the same in `remote-build.ts`.